addon/index.js

```diff
@@ -39,11 +39,14 @@ function validateArgs(obj, key) {
 export default function deepSet(obj, key, value) {
   runInDebug(() => validateArgs(obj, key));
   key.split('.').reduce((acc, currentKey, i, allKeys) => {
-    let currentValue = getWithDefault(acc, currentKey, {});
+    let currentValue =
+      get(acc, currentKey) === undefined ? {} : get(acc, currentKey);
```

Comment:
`get(acc, currentKey) ?? {};`

How about this?

Reply:
Agree, much readable. I had to upgrade and change some packages to use `??` operator, please take a look. Thanks!